### PR TITLE
[Submission] Hanssian

### PR DESCRIPTION
**Pull request by sevagh**
**HANSSIAN**

Dear mdx workshop committee,

I request a review for the following MDX submission:

### **Title:**  Music demixing with the sliCQ transform

**Author(s):**  Hanssian, Sevag

Link to my AICrowd profile (no team, solo submissions): https://www.aicrowd.com/participants/sevagh

**Category ***

* [x] `POSTERS` _(+Abstract: 250 Words+one page pdf)_
  > As a participant of the MDX challenge, please know that your poster would be automatically accepted after a minimal prescreening. Hence, don't forget to provide us with the link to your team in the AIcrowd website.
  > You are also welcome to submit us some unrelated abstract, in case your interest would be in presenting some method or idea you would just like to discuss with the participants passing by.  
  > => You *don't* need to include your poster in the PR. Just be ready to have it for D-day and show it to the audience through screen sharing.
* [ ] `LONG TALK`
  > A long talk will last about 20-30min, including questions. This can be about some recent paper you published, or may also present some original research and investigations you did. If you pick this category, please provide us with an extended abstract of the talk, or directly with the slides if you have them available, even in an unfinished state, at your convenience. 
* [ ] `DISCUSSIONS`
  > This category is intended to stimulate fruitful discussions and collaborations. If you pick it, it means you are willing to chair some open discussion with the participants, mainly around one particular aspect of music demixing research. For that purpose, you should provide a minimal abstract as well as a list of a few questions you would like the participants to discuss about. Don't hesitate to propose ideas, We will possibly contact you in case we identify several similar topics that were proposed!

**Preferred Timezone***

* [ ] A 11:00 PM (PDT) to 9:00 AM PDT
* [x] B 11:00 AM (PDT) to 9:00 PM PDT

  > The preferred timezone will help us to organize the day schedule. You should select at least one, but can perfectly select both if you don't care. We will organize the programme so that to maximally fit your preferences.
---

**ORGANISATION COMMITTEE (do not fill out)**

* [x] Editor acknowledgment
* [x] Reviewer 1 
* [ ] Reviewer 2
* [x] Review 1 decision [accept/reject]
* [ ] Review 2 decision [accept/reject]
* [ ] Editor decision [accept/reject]

### CODE REVIEW

**Comment by sevagh** [comment]
BTW, I noticed my spectrograms were kinda ugly:
* STFT was from MATLAB, with different colorbar/intensity label compared to the sliCQT
* The sliCQT title was missing a comma and capital B for bark
 
I have prepared a better version now, which I can update in the next submission:
image

File: paper.md [hunk 1]
```diff
@@ -1,139 +1,56 @@
...
+Music source separation, or music demixing, is the task of decomposing a song into its constituent sources, which are typically isolated instruments (e.g., drums, bass, and vocals). The Music Demixing Challenge^[<https://www.aicrowd.com/challenges/music-demixing-challenge-ismir-2021>] [@mdx21] was created to inspire new demixing research. Open-Unmix (UMX) [@umx], and the improved variant CrossNet-Open-Unmix (X-UMX) [@xumx], were included in the challenge as the baselines. Both models use the Short-Time Fourier Transform (STFT) as the representation of music signals.
 
+The time-frequency uncertainty principle states that the STFT of a signal cannot be maximally precise in both time and frequency [@gabor1946]. The tradeoff in time-frequency resolution can significantly affect music demixing results [@tftradeoff1]. Our proposed adaptation of Open-Unmix replaced the STFT with the sliCQT [@slicq], a time-frequency transform with varying time-frequency resolution. The resulting model, xumx-sliCQ^[<https://github.com/sevagh/xumx-sliCQ>] [@xumxslicq], was 5x smaller than UMX, but achieved lower demixing scores.
```
Comment (aliutkus) [inline paper.md:21]: achieved "better" demixing scores ?

File: paper.md [hunk 2]
```diff
@@ -1,139 +1,56 @@
...
-scientific explorations of forthcoming data releases from the *Gaia* mission
-[@gaia] by students and experts alike.
+In music demixing, the oracle estimator represents the theoretical upper limit of performance of a given strategy using ground truth signals. In Open-Unmix, the phase of the STFT is discarded, and the estimated magnitude STFT of the target is combined with the phase of the mix for the first estimate of the waveform. This is sometimes referred to as the "noisy phase" [@noisyphase1; @noisyphase2], described by \autoref{eq:noisyphaseoracle}.
+\begin{equation}\label{eq:noisyphaseoracle}
+\hat{X}_{\text{target}} = |X_{\text{target}}| \cdot \measuredangle{X_{\text{mix}}}
+\end{equation}
 
-## Mathematics
+The sliCQT parameters were chosen by maximizing the median SDR across all 4 targets from the noisy-phase waveforms in a 60-iteration random grid search on the validation set of MUDB18-HQ [@musdb18hq]. The sliCQT parameters of 262 frequency bins on the Bark scale between 32.9-22050 Hz achieved 7.42 dB, which surpassed the 6.23 dB of the STFT using the Open-Unmix default window and overlap of 4096 and 1024 respectively.
```
Comment (aliutkus) [inline paper.md:34]: this 7.42 dB SDR corresponds to the oracle performance or to your model ?

File: paper.md [hunk 3]
```diff
@@ -1,139 +1,56 @@
...
-\hat f(\omega) = \int_{-\infty}^{\infty} f(x) e^{i\omega x} dx
-\end{equation}
-and refer to \autoref{eq:fourier} from text.
+![Example of convolutional layers applied to a ragged sliCQT.\label{fig:ragged}](https://raw.githubusercontent.com/sevagh/mdx-submissions21/HANSSIAN/static-assets/xumx_slicq_pertarget.png){ width=100% }
 
-## Figures
+# Results
 
-Figures can be included like this:
+xumx-sliCQ was evaluated on the test set of MUSDB18-HQ and compared with the pretrained models of UMX and X-UMX. xumx-sliCQ achieved a median SDR of 3.6 dB, versus the 4.64 dB of UMX and 5.54 dB of X-UMX. xumx-sliCQ's inference was 2x slower than UMX, and the size of the trained model on disk was 5x smaller. The overall system architecture of xumx-sliCQ remains similar to UMX/X-UMX, as shown in \autoref{fig:blockdiagram}.
```
Comment (aliutkus) [inline paper.md:46]: here you report a performance that is worse that UMX. Indeed, higher is better for SDR. Is it really what was intended ?

**Review by aliutkus — COMMENTED** [review]
Nice twist where you replace the input representation to UMX and XUMX.
I'm confused whether the scores obtained are better or worse though, please make this clear

File: paper.md (hunk 2 above)
Comment (sevagh) [reply paper.md:34]: Yes, that's right, the noisy phase oracle using the chosen sliCQT parameters.

File: paper.md (hunk 1 above)
Comment (sevagh) [reply paper.md:21]: It was worse in its final performance - I couldn't beat UMX/X-UMX.

The overall story is:
1. sliCQT with chosen parameters has a better noisy-phase oracle performance than the STFT (7.42 dB vs. 6.23 dB)
    a. Which could be promising, _if_ somebody can find a good neural network architecture for it?
2. However, in the network I created and trained, xumx-sliCQ was *worse* than the originals

File: paper.md (hunk 3 above)
Comment (sevagh) [reply paper.md:46]: It was worse. It had maybe a higher theoretical ceiling from the noisy-phase oracle, but the real neural network was not great.

**Comment by sevagh** [comment]
To clarify, the noisy phase oracle showed a _potential_ increase in SDR by using the sliCQT.

However, for the real network, I only managed to build an empirical NN layout as a proof of concept, where the end result was worse than UMX/XUMX.

I still think the sliCQT is a promising idea if a better NN architecture is chosen, that's why I went ahead with writing the paper.

**Comment by aliutkus** [comment]
To all participants:
please note that you should update your submission by November 6th 23:00 UTC

Thank you very much

**Comment by sevagh** [comment]
Very cool, I see myself in the Poster B session! Does this mean it's accepted? It's my first academic publication :partying_face:

**Comment by faroit** [comment]
@sevagh yes you are 👏  The MDX ISMIR program will be finalized today and you will see your paper in the midspace platform. Also the proceeding will be published on our website by end of next week

I suggest you also put this on arxiv, its some nice overview

**Comment by faroit** [comment]
👏 The submission has been published on midspace (the ISMIR conference platform). The submissions will also be hosted on the workshop website tomorrow. 

The schedule can be seen there or [here](https://docs.google.com/spreadsheets/d/1PaksFcHWUfNPMg-XloA3CJtrvb9uMU0erCFVyvFi9B4/edit?usp=sharing). Please let us know if there is anything missing. Otherwise we see each other on Friday/Saturday!

**Comment by faroit** [comment]
@sevagh can you please upload your poster to midspace or send this to me?

**Comment by sevagh** [comment]
@faroit oops, sorry about that - I attached it in our MidSpace chat when we spoke earlier today

[mdx21-poster.pdf](https://github.com/mdx-workshop/mdx-submissions21/files/7530860/mdx21-poster.pdf)

**Comment by faroit** [comment]
@sevagh sorry my fault. thanks! your poster is all set on gather town now

**Comment by sevagh** [comment]
If I want to put this on arXiv, should I put it in "Computer Science -> Sound", or "EECS -> Audio and Speech Processing"? Also, can you be my arXiv endorser?

**Comment by sevagh** [comment]
I would also like to formalize the several ideas I have that I think would make a big difference in the performance of xumx-sliCQ - I talked about them at my poster session, based on this document I created: https://gitlab.com/sevagh/xumx_slicq_extra/-/tree/main/sliceq22-ideas

Should I write a bigger paper for arXiv, containing the contents of my MDX21 extended abstract + a future ideas section? Or is it best to submit this extended abstract as-is?

Basically I think there is a good flow of current design choices of xumx-sliCQ + future improvements:

* Section 0: motivation, time-frequency uncertainty, tradeoff with STFT window size, etc.
* Section 1: sliCQT parameter search
    * **Current**: maximize noisy-phase SDR search, using a single sliCQT for all 4 targets
    * **Future 1**: improved phase of 2014 sliCQT paper
    * **Future 2**: use different sliCQT for all 4 targets (different size of outputs that cannot be summed together for cross-target loss), but using the STFT domain to compute frequency-domain loss (thus avoiding the different size problem)
    * **Future 3**: ridge analysis for objectively measuring the best possible sliCQT per target (which can be even more relevant in the future with, as you mentioned, electric guitar as a source, etc.)
* Section 2: neural network design
    * **Current**: looping over time-frequency buckets and creating mini-neural networks per
    * **Future**: improved rasterized single matrix form of 2014 sliCQT paper, such that the network learns from all the frequency bins together, and making it more like a simple STFT-based network
* Section 3: non-invertible 50% overlap-add procedure
    * **Current**: extra ConvTranspose(stride=2) layer to grow back to the original size - joint training of xumx-sliCQ includes the music demixing, and learned de-overlap layer
    * **Future**: separate neural network to learn DeOverlap, and then use pre-trained DeOverlapNet in the music demixing network

**Comment by faroit** [comment]
@sevagh I will reply to this next week.

**Comment by sevagh** [comment]
Did you have any more thoughts on this? I think for simplicity, I should submit the same paper as MDX21. I can save my future ideas for MDX22.

**Comment by sevagh** [comment]
@faroit I have prepared the arXiv submission. It's 98% similar to the MDX 21 paper, but with some improved diagrams and slightly better wording around some key sliCQT concepts.

However, I need an endorsement for cs.SD (and my own research lab doesn't have anybody who can do that).

**Comment by faroit** [comment]
@aliutkus can you take care of this?

**Comment by sevagh** [comment]
I got endorsed by @KinWaiCheuk and submitted it
